src/platform/console/unix_console.cpp

```diff
@@ -82,15 +81,15 @@ class mp::WindowChangedSignalHandler
     mp::AutoJoinThread signal_handling_thread;
 };
 
-mp::UnixConsole::UnixConsole(ssh_channel channel)
-    : interactive{mcp::is_tty()}, handler{std::make_unique<WindowChangedSignalHandler>(channel)}
+mp::UnixConsole::UnixConsole(ssh_channel channel, Terminal* term)
+    : term{term}, handler{std::make_unique<WindowChangedSignalHandler>(channel, term->cout_fd())}
 {
     setup_console();
 
-    if (interactive)
+    if (term->cout_is_tty())
     {
         ssh_channel_request_pty(channel);
-        change_ssh_pty_size(channel);
+        change_ssh_pty_size(channel, term->cout_fd());
```

Comment:
A bit of a side question, how come `cerr` does not receive the same treatment as `cout`?

Reply:
I didn't see it being used much. Totally could, if we want consistency

Reply:
Right, I wondered why it wasn't used, but that is probably a separate thing